source/common/http/conn_manager_impl.cc

```diff
@@ -854,13 +856,29 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
   std::list<ActiveStreamDecoderFilterPtr>::iterator entry;
   auto trailers_added_entry = decoder_filters_.end();
   const bool trailers_exists_at_start = request_trailers_ != nullptr;
+  // Indicates if to check the filter's iteration is stopped for all frame types.
+  bool check_stop_all = true;
   if (!filter) {
     entry = decoder_filters_.begin();
   } else {
-    entry = std::next(filter->entry());
+    if ((*(filter->entry()))->iterate_from_current_filter_) {
```

Comment:
This logic block is repeated multiple times, and I'm having a really hard time understanding the `check_stop_all` logic. Is it possible to somehow consolidate all of this handling into common helper functions with substantially more comments on what is going on? If not, can we get more comments in all of the call sites on what we are doing with `check_stop_all` ?

Reply:
Sorry about the confusion and good catch! I think after introducing iterate_from_current_filter_,  the iteration state will be reset before calling doData and doTrailers, and we don't need check_stop_all, which is used to make sure we skip the state check if the filter is already in StopAll state.